src/python/pants/backend/python/goals/pytest_runner_integration_test.py

```diff
@@ -281,31 +281,31 @@ def test():
     assert f"{PACKAGE}/tests.py ." in result.stdout_simplified_str
 
 
-@skip_unless_python37_and_python39_present
+@skip_unless_python38_and_python39_present
 def test_uses_correct_python_version(rule_runner: PythonRuleRunner) -> None:
     rule_runner.write_files(
         {
             f"{PACKAGE}/tests.py": dedent(
                 """\
                 def test() -> None:
-                  y = (x := 5)
+                  y = {} | {}
```

Comment:
We use this pattern in a few places where the tests depended the walrus operator change.  Where possible I tried to find an similar 3.8-->3.9 syntax change, but I don't think relying on syntax changes in every CPython release is a scalable pattern.